src/providers/cloudflare-stream.js

```diff
@@ -0,0 +1,166 @@
+const { ok } = require('node:assert');
+const crypto = require('node:crypto');
+const AbstractFileTransfer = require('ms-files-transport');
+const Cloudflare = require('cloudflare');
+const { HttpStatusError } = require('common-errors');
+
+const toBase64 = (value) => Buffer.from(value).toString('base64');
+const nowPlusSeconds = (seconds) => (new Date(Date.now() + 1000 * seconds)).toISOString();
+const nowPlus30Days = () => nowPlusSeconds(2592000);
+const NotImplementedHttpError = new HttpStatusError(501, 'Method \'copy\' is not implemented');
+
+const arrayBufferToBase64Url = (buffer) => btoa(String.fromCharCode(...new Uint8Array(buffer)))
+  .replace(/=/g, '')
+  .replace(/\+/g, '-')
+  .replace(/\//g, '_');
```

Comment:
extremely expensive, just use Buffer.toString('base64url') - there is an encoding support for that in node.js